### PR TITLE
flutter upgrade version

### PR DESCRIPTION
**Pull request by bhargavraviya**
Flutter Upgrade from scratch

- Flutter Version Upgrade 
- New version code changes
- Android files changes
- Package upgrade
- Android kotlin version upgrade

please check it and merge @YC

### CODE REVIEW

**Comment by YC** [comment]
Hi @bhargavraviya, thanks for your PR. 
While I agree with most of the changes, I would prefer to not review nor accept this in one single PR, given that it touches multiple separate concerns at once (e.g. refactoring, removal of ios, changes to android packaging, changes to flutter packages). 

If you're okay with splitting this into multiple PRs, I'd be more than happy to review them separately. 
In this case, I'd welcome separate PRs for:
- Code formatting (e.g. reordering of lines)
- Flutter Version upgrades
- Routing related fixes
- Android project changes
- Other refactoring

Otherwise, I'd be happy to integrate your changes through a set of commits (giving you credit for the changes as co-contributor). 

Happy to discuss further!

File: .gitignore [hunk 1]
```diff
@@ -29,7 +30,7 @@
 .packages
 .pub-cache/
 .pub/
-build/
+/build/
```
Comment (YC) [inline .gitignore:32]: `build/` should cover `/build/`?

File: .metadata [hunk 2]
```diff
@@ -4,7 +4,24 @@
...
 version:
-  revision: 985ccb6d14c6ce5ce74823a4d366df2438eac44f
-  channel: beta
+  revision: "2663184aa79047d0a33a14a3b607954f8fdd8730"
+  channel: "stable"
 
 project_type: app
+
+# Tracks metadata for the flutter migrate command
+migration:
```
Comment (YC) [inline .metadata:13]: Didn't know about `flutter migrate`. 
I've previously did updates to the `android/` and `ios/` projects manually. 
Thanks for bringing it to my attention.

File: android/app/build.gradle [hunk 3]
```diff
@@ -5,43 +5,29 @@ plugins {
...
-
-def flutterVersionName = localProperties.getProperty("flutter.versionName")
-if (flutterVersionName == null) {
-    flutterVersionName = "1.0"
-}
-
 android {
     namespace = "net.steventang.another_authenticator"
-    compileSdk = 34
+    compileSdk = flutter.compileSdkVersion
```
Comment (YC) [inline android/app/build.gradle:10]: I think I've had a compile error with `flutter.compileSdkVersion` previously. 
May need to double check this.

File: android/app/build.gradle [hunk 4]
```diff
@@ -56,6 +42,3 @@ android {
 flutter {
     source = "../.."
 }
-dependencies {
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
```
Comment (YC) [inline android/app/build.gradle:60]: Can this be removed?
Was listed as requirement in:
https://pub.dev/packages/barcode_scan2

File: android/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -31,22 +31,26 @@
...
+            <action android:name="android.intent.action.PROCESS_TEXT"/>
+            <data android:mimeType="text/plain"/>
+        </intent>
         <!-- Launching URLs -->
         <intent>
             <action android:name="android.intent.action.VIEW" />
             <data android:scheme="https" />
         </intent>
     </queries>
+    <uses-permission android:name="android.permission.INTERNET" />
```
Comment (YC) [inline android/app/src/main/AndroidManifest.xml:54]: Is this needed?
From privacy perspective, I'd prefer if this app doesn't require Internet.

File: android/app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -31,22 +31,26 @@
...
+    <uses-permission android:name="android.permission.CAMERA" />
```
Comment (YC) [inline android/app/src/main/AndroidManifest.xml:55]: May need to look into this further. 
It'd be ideal if this app can work without camera, and only accesses camera when needed. 
https://developer.android.com/media/camera/camera-deprecated/camera-api

File: android/build.gradle [hunk 7]
```diff
@@ -1,14 +1,3 @@
-buildscript {
-    ext.kotlin_version = '1.7.10'
-    repositories {
-        google()
-        mavenCentral()
-    }
-
-    dependencies {
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
```
Comment (YC) [inline android/build.gradle:9]: Same comment as the other gradle file.

**Review by YC — CHANGES_REQUESTED** [review]

File: android/app/build.gradle (hunk 3 above)
Comment (bhargavraviya) [reply android/app/build.gradle:10]: please change in android/local.properties

```
flutter.minSdkVersion=23
flutter.targetSdkVersion=34
flutter.compileSdkVersion=35
```

@YC

File: android/app/build.gradle (hunk 4 above)
Comment (bhargavraviya) [reply android/app/build.gradle:60]: new flutter change location in settings.gradle like 
```
  id "org.jetbrains.kotlin.android" version "1.9.23" apply false
```
@YC

File: android/app/src/main/AndroidManifest.xml (hunk 5 above)
Comment (bhargavraviya) [reply android/app/src/main/AndroidManifest.xml:54]: Yes, it is needed. Please update it because it is required to fetch the 2FA data.

**Comment by YC** [comment]
Thanks again. 
Moving reviews to #15 and #16.

**Pull request closed without merging**